app/src/main/java/com/cornellappdev/android/eatery/components/WaitTimesFragment.java

```diff
@@ -6,87 +6,130 @@
+
     private void setupWaitTimesChart() {
+        mWaitTimesChart.setNoDataText("No wait times available.");
+        mWaitTimesChart.setNoDataTextColor(getResources().getColor(R.color.secondary));
+        mWaitTimesChart.setNoDataTextTypeface(Typeface.SANS_SERIF);
         mWaitTimesChart.setDescription(null);
         mWaitTimesChart.getLegend().setEnabled(false);
         mWaitTimesChart.setDoubleTapToZoomEnabled(false);
         mWaitTimesChart.setPinchZoom(false);
-        mWaitTimesChart.animateY(2000);
-        mWaitTimesChart.setExtraBottomOffset(24f);
-        mWaitTimesChart.setExtraTopOffset(48f);
         mWaitTimesChart.getRendererXAxis().getPaintAxisLabels().setTextAlign(Paint.Align.LEFT);
+        mWaitTimesChart.setViewPortOffsets(0, 100f, 0, 0);
+
+        this.setupWaitTimesChartAxis();
+        this.setupWaitTimesData();
+
+        // Set up wait times marker label.
+        mWaitTimesMarkerView = new WaitTimesMarkerView(getContext(), R.layout.wait_times_marker_view);
+        mWaitTimesChart.setMarker(mWaitTimesMarkerView);
+        highlightCurrentHour();
+    }
+
+    private void setupWaitTimesData() {
         mWaitTimesChart.setOnChartValueSelectedListener(new OnChartValueSelectedListener() {
-//          TODO (yanlam): update with wait times from backend
             @Override
             public void onValueSelected(Entry e, Highlight h) {
-
+                Swipe s = mSwipeData.get((int) e.getX());
+                mWaitTimesMarkerView.updateMarkerLabel(e, s);
             }
-
             @Override
             public void onNothingSelected() {
-
+                highlightCurrentHour();
             }
         });
 
-        this.setupWaitTimesChartAxis();
-
+        // Use wait times data from backend.
         List<BarEntry> entries = new ArrayList<>();
-//        TODO (yanlam): pull wait times from backend and make bar entries dynamic.
-        entries.add(new BarEntry(0, 2f));
-        entries.add(new BarEntry(1f, 30f));
-        entries.add(new BarEntry(2f, 10f));
-        entries.add(new BarEntry(3f, 20f));
-        entries.add(new BarEntry(4f, 30f));
-        entries.add(new BarEntry(5f, 40f));
-        entries.add(new BarEntry(6f, 40f));
-        entries.add(new BarEntry(7f, 35f));
-        entries.add(new BarEntry(8f, 40f));
-        entries.add(new BarEntry(9f, 30f));
-        entries.add(new BarEntry(10f, 2f));
-        entries.add(new BarEntry(11f, 5f));
-        entries.add(new BarEntry(12f, 10f));
-        entries.add(new BarEntry(13f, 40f));
-        entries.add(new BarEntry(14f, 40f));
-        entries.add(new BarEntry(15f, 20f));
-        entries.add(new BarEntry(16f, 2f));
-        entries.add(new BarEntry(17f, 2f));
-        entries.add(new BarEntry(18f, 10f));
-        entries.add(new BarEntry(19f, 2f));
-        entries.add(new BarEntry(20f, 2f));
+        for (int i = 0; i < mSwipeData.size(); i++) {
+            entries.add(new BarEntry(i, (float) mSwipeData.get(i).swipeDensity));
+        }
```

Comment:
@connorreinhold  what do you think: does this block belong in the fragment class or a presenter?

Reply:
My opinion here is that these lines are creating a list of BarEntry's which are specific to the display, and since we are just populating a graph here I think it's fine to not have this in a presenter. Then again, could just be me being lazy